core/src/main/java/io/seata/core/rpc/RemotingClient.java

```diff
@@ -33,43 +34,42 @@
 public interface RemotingClient {
 
     /**
-     * Send msg with response object.
+     * client send sync request.
+     * In this request, if {@link NettyClientConfig#isEnableClientBatchSendRequest} is enabled,
+     * the message will be sent in batches.
      *
-     * @param msg     the msg
-     * @param timeout the timeout
-     * @return the object
-     * @throws TimeoutException the timeout exception
+     * @param msg transaction message {@link io.seata.core.protocol}
+     * @return server result message
+     * @throws TimeoutException
      */
-    Object sendMsgWithResponse(Object msg, long timeout) throws TimeoutException;
+    Object sendSyncRequest(Object msg) throws TimeoutException;
 
     /**
-     * Send msg with response object.
+     * client send sync request.
      *
-     * @param serverAddress the server address
-     * @param msg           the msg
-     * @param timeout       the timeout
-     * @return the object
-     * @throws TimeoutException the timeout exception
+     * @param channel client channel
+     * @param msg     transaction message {@link io.seata.core.protocol}
+     * @return server result message
+     * @throws TimeoutException
      */
-    Object sendMsgWithResponse(String serverAddress, Object msg, long timeout) throws TimeoutException;
+    Object sendSyncRequest(Channel channel, Object msg) throws TimeoutException;
```

Comment:
should there be a channel parameter?

Reply:
At that time, why the channel invaded the RemotingClient interface was also a helpless choice, because when called in many places, the various parameters were not uniform, it was difficult to remove the dependency of the channel. At that time, my idea was,
1. RemotingClient as a top-level interface, when other rpc types are added in the future, their interfaces can be added to this top-level interface
2. Remove the netty-related dependencies and make a RemotingClient that is truly irrelevant to the specific rpc type
In the end I chose the first and easiest method to refactor.


I would like to add a pr later to improve it.